modules/programs/ghostty.nix

```diff
@@ -0,0 +1,171 @@
+
+        See <https://ghostty.org/docs/config/reference> for more information.
+      '';
+    };
+
+    themes = lib.mkOption {
+      type = lib.types.attrsOf keyValue.type;
+      default = { };
+      example = {
+        catppuccin-mocha = {
+          palette = [
+            "0=#45475a"
+            "1=#f38ba8"
+            "2=#a6e3a1"
+            "3=#f9e2af"
+            "4=#89b4fa"
+            "5=#f5c2e7"
+            "6=#94e2d5"
+            "7=#bac2de"
+            "8=#585b70"
+            "9=#f38ba8"
+            "10=#a6e3a1"
+            "11=#f9e2af"
+            "12=#89b4fa"
+            "13=#f5c2e7"
+            "14=#94e2d5"
+            "15=#a6adc8"
+          ];
+          background = "1e1e2e";
+          foreground = "cdd6f4";
+          cursor-color = "f5e0dc";
+          selection-background = "353749";
+          selection-foreground = "cdd6f4";
+        };
+      };
+      description = ''
+        Custom themes written to {file}`$XDG_CONFIG_HOME/ghostty/themes`.
+
+        See <https://ghostty.org/docs/features/theme#authoring-a-custom-theme> for more information.
+      '';
+    };
+
+    clearDefaultKeybinds = lib.mkEnableOption "" // {
+      description = "Whether to clear default keybinds.";
+    };
+
+    installVimSyntax =
+      lib.mkEnableOption "installation of Ghostty configuration syntax for Vim";
+
+    installBatSyntax =
+      lib.mkEnableOption "installation of Ghostty configuration syntax for bat"
+      // {
+        default = true;
+      };
+
+    enableBashIntegration = lib.mkEnableOption ''
+      bash shell integration.
+
+      This is ensures that shell integration works in more scenarios, such as switching shells within Ghostty.
+      But it is not needed to have shell integration.
+      See <https://ghostty.org/docs/features/shell-integration#manual-shell-integration-setup> for more information
+    '';
+
+    enableFishIntegration = lib.mkEnableOption ''
+      fish shell integration.
+
+      This is ensures that shell integration works in more scenarios, such as switching shells within Ghostty.
+      But it is not needed to have shell integration.
+      See <https://ghostty.org/docs/features/shell-integration#manual-shell-integration-setup> for more information
+    '';
+
+    enableZshIntegration = lib.mkEnableOption ''
+      zsh shell integration.
+
+      This is ensures that shell integration works in more scenarios, such as switching shells within Ghostty.
+      But it is not needed to have shell integration.
+      See <https://ghostty.org/docs/features/shell-integration#manual-shell-integration-setup> for more information
+    '';
+  };
+
+  config = lib.mkIf cfg.enable (lib.mkMerge [
+    {
+      home.packages = [ cfg.package ];
+
+      programs.ghostty.settings = lib.mkIf cfg.clearDefaultKeybinds {
+        keybind = lib.mkBefore [ "clear" ];
+      };
+
+      xdg.configFile = lib.mkMerge [
+        {
+          "ghostty/config" = lib.mkIf (cfg.settings != { }) {
```

Comment:
On macOS, this config file is in `~/Library/Application Support/com.mitchellh.ghostty/config`. Please add a branch for that.

Reply:
This was already discussed in: https://github.com/nix-community/home-manager/pull/6235#discussion_r1898657805. Ghostty will look for both

Reply:
Ah, sorry. GitHub UI doesn't let you search through comments easily (or at all). Maybe add a comment in code to reference that to avoid any confusion in the future?

Reply:
> Ah, sorry. GitHub UI doesn't let you search through comments easily (or at all). Maybe add a comment in code to reference that to avoid any confusion in the future?

No problem. I don't really think a code comment is needed. But I might add one if I modify the module again.

Reply:
https://ghostty.org/docs/config#file-location

its also very well documented on the website